File: 2024/Jan/day12.cpp
class Solution
{
    public:
    
    // Function to reverse first k elements of a queue.
    queue<int> modifyQueue(queue<int> q, int k) {
        // add code here.
        stack<int> s;
        for(int i=0;i<k;i++)
        {
            s.push(q.front());
            q.pop();
        }
        
        while(!s.empty())
        {
            q.push(s.top());
            s.pop();
        }
        int n=q.size();
        for(int i=1;i<=n-k;i++)
        {
            int ans=q.front();
            q.pop();
            q.push(ans);
        }
        return q;
    }
};

File: 2024/Feb/day12.py
class Solution:
    def sequence(self, n):
        # code here
        M = 10**9+7
        ans = 0
        start = 1
        for i in range(1,n+1):
            v = 1
            while i > 0:
                v = v*start%M
                start += 1
                i -= 1
            ans = (ans+v)%M
        return ans
